### PR TITLE
Fixed Grammar and Spelling Typo's

### PR DESCRIPTION
**Pull request by MrKrishnaAgarwal**
**Fixed some grammatical issues for clarity and ease of readability.**

### CODE REVIEW

**Review by francofgp — COMMENTED** [review]
looks good to me

**Comment by MrKrishnaAgarwal** [comment]
**Thanks**

File: docs/project/goals.md [hunk 1]
```diff
@@ -79,14 +79,14 @@ expectations around conduct and behavior need to be clearly articulated both to
 set expectations for people joining, and to help remind and anchor us on
 consistent standards. It is also important that we hold ourselves accountable to
 these expectations and have real and meaningful mechanisms to moderate the
-community. When behavior steps outside of our expectations, we need tools,
-process, and policy for how we will recognize and correct it.
+community. When behavior steps outside of our expectations, we need tools, 
+processes, and policy for how we will recognize and correct it.
```
Comment (geoffromer) [inline docs/project/goals.md]: Should this be "policies" as well, for consistency?

File: docs/project/goals.md [hunk 2]
```diff
@@ -79,14 +79,14 @@ expectations around conduct and behavior need to be clearly articulated both to
...
 
 **An open, inclusive process for Carbon changes.** The community needs to be
 able to effectively engage in the direction and evolution of the project and
 language, while keeping the process efficient and effective. That means we need
 an open, inclusive process where everyone feels comfortable participating.
-Community members should understand how and why decisions are made, and have the
+Community members should understand how and why decisions are made and have the
```
Comment (geoffromer) [inline docs/project/goals.md]: This change feels like a slight regression to me. Both versions are correct, but I find the old one easier to parse.

File: docs/project/goals.md [hunk 3]
```diff
@@ -130,17 +130,17 @@ expected to read through the specification to ramp up with Carbon. User guides
...
-out-of-the-box in order to encourage adoption of Carbon at scale where it can
+**Compelling adoption tooling.** We want to provide a compelling suite of tools 
+out of the box to encourage the adoption of Carbon at a scale where it can
 augment existing C++ codebases. For example, we expect a C++ -> Carbon code
 translator will be important.
 
 **Tooling for updating code when Carbon evolves.** As Carbon evolves over time,
 we expect to provide tooling to help automate and scale migrating existing
-Carbon code to the new version. The goal is to enable more rapid evolution of
+Carbon code to the new version. The goal is to enable a more rapid evolution of
```
Comment (geoffromer) [inline docs/project/goals.md:140]: This doesn't sound quite right to me -- we're talking here about "evolution" in the sense of an ongoing process of change, but using "evolution" as a count noun with a count of one makes it sound like we're talking about a single event.

Is there a problem with the old phrasing that you're trying to solve here?

File: docs/project/goals.md [hunk 4]
```diff
@@ -167,7 +167,7 @@ Many languages share subsets of these goals, but what distinguishes Carbon is
 their combination. Where it is necessary to make tradeoffs between these goals,
 we intend to prioritize them in this order.
 
-Each goal is broad, and has several facets to consider when making decisions.
+Each goal is broad and has several facets to consider when making decisions.
```
Comment (geoffromer) [inline docs/project/goals.md]: Here again, removing the comma feels like a slight regression.

File: docs/project/goals.md [hunk 5]
```diff
@@ -250,7 +249,7 @@ extending it are inherently hostile towards long-term maintenance of software.
 not get the design of most language features correct on our first, second, or
 73rd try. As a consequence, there must be a built-in plan and ability to move
 Carbon forward at a reasonable pace and with a reasonable cost. Simultaneously,
-an evolving language must not leave software behind to languish, but bring
+an evolving language must not leave software behind to languish, instead bringing
```
Comment (geoffromer) [inline docs/project/goals.md]: I don't think "bringing" is grammatical here -- it parallels "leave", so it should be conjugated the same way.

File: docs/project/goals.md [hunk 6]
```diff
@@ -199,7 +198,7 @@ regular, idiomatic code usually results in high performance.
 
 **Code should perform predictably.** The reader and writer of code should be
 able to easily understand its expected performance, given sufficient background
-knowledge of the environment in which it will run. This need not be precise, but
+knowledge of the environment in which it will run. This need not be precise, but
```
Comment (geoffromer) [inline docs/project/goals.md]: Is this change intentional?

**Review by geoffromer — COMMENTED** [review]
Few if any of these cases look like typos to me, so "Fixed Grammar and Spelling Typo's" seems inaccurate. Maybe something like "Style tweaks and clarifications"?

**Comment by MrKrishnaAgarwal** [comment]
@geoffromer 
Sorry for the problems caused,
I tried my best to improve it.

Also I don't know why the pre-commit is failing
Can you merge the pull request?

**Comment by geoffromer** [comment]
You marked my comments resolved, but I don't see any changes in the pull request. Do you need to push from your local client?

> @geoffromer Sorry for the problems caused, I tried my best to improve it.

You didn't cause any problems, and there's no need to apologize. This is what code review is for :-)

> Also I don't know why the pre-commit is failing Can you merge the pull request?

It looks like minor formatting issues. Have you [set up pre-commit](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#pre-commit) in your local client? If so, you can run it to automatically fix the formatting.

**Comment by MrKrishnaAgarwal** [comment]
> It looks like minor formatting issues. Have you [set up pre-commit](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#pre-commit) in your local client? If so, you can run it to automatically fix the formatting.

I have not set up, is it mandatory

**Comment by MrKrishnaAgarwal** [comment]
> You marked my comments resolved, but I don't see any changes in the pull request. Do you need to push from your local client?
 I have pushed the changes.

**Comment by MrKrishnaAgarwal** [comment]
Please see the changes now, and if you approve, the pull request can be merged ✅.
image

File: docs/project/goals.md (hunk 3 above)
Comment (geoffromer) [reply docs/project/goals.md:140]: This still looks unresolved.

**Comment by geoffromer** [comment]
> > It looks like minor formatting issues. Have you [set up pre-commit](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#pre-commit) in your local client? If so, you can run it to automatically fix the formatting.
> 
> I have not set up, is it mandatory

It's not mandatory to run the pre-commit locally, but we can't merge the PR unless it passes the pre-commit on GitHub, which implements the same checks. Local pre-commit runs will give you much faster feedback, and in a lot of cases (including the check that this PR is currently failing) it will automatically edit your files to fix the problem.

**Comment by MrKrishnaAgarwal** [comment]
How should I run the pre commit in local, and how to push the changes after that.

I've also allowed "edits by maintainers" you can also edit the file..

**Comment by MrKrishnaAgarwal** [comment]
#1795 

One more pull request, I ran pre commit

Try in this one

**Comment by geoffromer** [comment]
> How should I run the pre commit in local, and how to push the changes after that.

See the documentation [here](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/contribution_tools.md#pre-commit).

> I've also allowed "edits by maintainers" you can also edit the file..

I would really appreciate it if you could make the necessary changes yourself.

Also, in case you missed it, there's still an unresolved comment on `goals.md`.

**Comment by MrKrishnaAgarwal** [comment]
@geoffromer 
Why did this pull request closed unexpectedly?

Have you closed this?

File: docs/project/goals.md (hunk 3 above)
Comment (MrKrishnaAgarwal) [reply docs/project/goals.md:140]: The pull request closed automatically, I'll not be able to change anything now.

**Comment by jonmeow** [comment]
The neutral "close" message typically means it was closed by GitHub for some indirect reason. In this case, it looks like the underlying fork may have been deleted, so the PR became invalid. Clicking on the source link (https://github.com/MrKrishnaAgarwal/carbon-lang/tree/patch-1) gives a 404.

**Comment by MrKrishnaAgarwal** [comment]
Thanks a lot 
I understood
